Add: ability to include generated images in the conversation.

Fraggle opened the pull request:
## Description

Added support to include images in the conversation.
- Expand function call return value type to include structured output (supported by claude and required for it to see images).
- Make images "includable".
- Improve handling of tool results by supporting structured output for claude.
- OpenAI do not support structured output but figure out what to do with the image url in the text content.

## Tests

Did several tests, also checked that it still directly read images when they are submitted by the user.

## Risk

Medium, but the API is backward compatible.

## Deploy Plan

Deploy `core` then `front`

Fraggle commented on core/src/providers/anthropic.rs:
```diff
@@ -64,10 +64,27 @@ impl ToString for AnthropicChatMessageRole {
+#[serde()]
+struct AnthropicContentToolResultContent {
+    r#type: AnthropicContentToolResultContentType,
+    #[serde(skip_serializing_if = "Option::is_none")]
```
this one lost me an hour

flvndvd replied:
Serde is 🔥

spolu commented on core/src/providers/anthropic.rs:
```diff
@@ -64,10 +64,27 @@ impl ToString for AnthropicChatMessageRole {
+#[serde()]
+struct AnthropicContentToolResultContent {
+    r#type: AnthropicContentToolResultContentType,
+    #[serde(skip_serializing_if = "Option::is_none")]
```
ignore

spolu commented on core/src/providers/anthropic.rs:
```diff
@@ -64,10 +64,27 @@ impl ToString for AnthropicChatMessageRole {
+#[serde()]
+struct AnthropicContentToolResultContent {
+    r#type: AnthropicContentToolResultContentType,
+    #[serde(skip_serializing_if = "Option::is_none")]
```
ignore

spolu reviewed (approved):
If I understand correctly the goal is to model properly generated images as tool output when feeding back conversation state to the model.

What's the story for providers that don't support it? I don't see a google diff here?

Overall LGTM deferring final review to @flvndvd or @fontanierh

Fraggle commented:
> If I understand correctly the goal is to model properly generated images as tool output when feeding back conversation state to the model.

Exactly, currently they are presented as part of the serialized JSON output which isn't handled correctly by Claude. Claude expect a tool output "content" to be either a string or a content block with type text or image (like a user message content block) => [doc](https://docs.anthropic.com/en/docs/build-with-claude/tool-use/overview#handling-tool-use-and-tool-result-content-blocks) (bullet point 3.)

> What's the story for providers that don't support it? I don't see a google diff here?

I expected rust to be my guard since I changed the type of the global "FunctionCallResult" message.
It did error in mistral and the open ai compatible providers helper which I fixed. I imagine that google is using the open ai compatible layer but I'll double check.

Fraggle commented:
> I imagine that google is using the open ai compatible layer but I'll double check.

[Yep, it uses the open ai compatible helpers.](https://github.com/dust-tt/dust/blob/a9b633c73592a7c313592d52afbcae598bf158cb/core/src/providers/google_ai_studio.rs#L146)

flvndvd commented on front/lib/actions/mcp.ts:
```diff
@@ -510,10 +510,15 @@ export class MCPConfigurationServerRunner extends BaseActionConfigurationServerR
         isSupportedFileContentType(c.resource.mimeType)
       ) {
+        const fileName =
+          "name" in c.resource && typeof c.resource.name === "string"
```
`nit` Typeguard
